source/slang/slang-diagnostic-defs.h

```diff
@@ -317,44 +317,46 @@ DIAGNOSTIC(30600, Error, varWithoutTypeMustHaveInitializer, "a variable declarat
 // 307xx: parameters
 DIAGNOSTIC(30700, Error, outputParameterCannotHaveDefaultValue, "an 'out' or 'inout' parameter cannot have a default-value expression");
 
-DIAGNOSTIC(39999, Error, expectedIntegerConstantWrongType, "expected integer constant (found: '$0')")
-DIAGNOSTIC(39999, Error, expectedIntegerConstantNotConstant, "expression does not evaluate to a compile-time constant")
-DIAGNOSTIC(39999, Error, expectedIntegerConstantNotLiteral, "could not extract value from integer constant")
+DIAGNOSTIC(30701, Error, expectedIntegerConstantWrongType, "expected integer constant (found: '$0')")
+DIAGNOSTIC(30702, Error, expectedIntegerConstantNotConstant, "expression does not evaluate to a compile-time constant")
+DIAGNOSTIC(30703, Error, expectedIntegerConstantNotLiteral, "could not extract value from integer constant")
 
```

Comment:
Hmm... The comment says `307xx` is for parameter-related errors, but your edits here have extended that range to include a *lot* of unrelated error messages (including error messages not related to your change).

I'd really prefer that we keep unrelated "cleanup" stuff out of PRs, partly to avoid merge conflicts (there is a non-zero chance that our two open PRs will end up with merge conflicts because of these edits), and partly to avoid having to spend time on design discussion that isn't really about the feature being worked on.

In this case, all the `39999` errors were indeed waiting for a good time to give them more permanent numbers. One problem is that our diagnostics need a lot of scrubbing (there are a bunch of diagnostics that are completely unused), and the numbering realistically needs a major rethink/reorg.

The intention with the header comments like `307xx : parameters` was to start reserving ranges of diagnostic numbers for related messages, so that we could keep the overall space somewhat sparse and not have to worry as much about fitting in a new message near the ones it relates to.

I would have greatly preferred to see this PR reserve `308xx` for literal-related diagnostics (or maybe even numeric literals), and put all its diagnostics in that range without rewriting any of the existing ones.

Reply:
Ok - will revert. Added a comment, so it doesn't just seem like an oversight.